src/Common/Request/Query_Vars.php

```diff
@@ -0,0 +1,253 @@
+<?php
+/**
+ * Request query vars filter and sanitization.
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Request
+ */
+
+declare(strict_types=1);
+
+namespace TEC\Common\Request;
+
+/**
+ * Class Query_Vars
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Request
+ */
+class Query_Vars {
+	/**
+	 * The GET superglobal.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public const SUPERGLOBAL_GET = '_GET';
+
+	/**
+	 * The POST superglobal.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public const SUPERGLOBAL_POST = '_POST';
+
+	/**
+	 * The REQUEST superglobal.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public const SUPERGLOBAL_REQUEST = '_REQUEST';
+
+	/**
+	 * The allowed superglobals check list.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string>
+	 */
+	private const ALLOWED_SUPERGLOBALS = [
+		self::SUPERGLOBAL_GET,
+		self::SUPERGLOBAL_POST,
+		self::SUPERGLOBAL_REQUEST,
+	];
+
+	/**
+	 * Register hooks.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function register(): void {
+		do_action( 'tec_request_query_vars_registered' );
```

Comment:
After you make it a controller you will be able to specify `public static string $registration_action = 'tec_request_query_vars_registered';`